resources/builder.inc.sh

```diff
@@ -400,9 +400,19 @@ _builder_execute_child() {
     fi
   done
 
-  "$script" $action \
+  # If the current build is a dependency build, pass the dependency state on to
+  # the child build, so that we don't unnecessarily rebuild children (#11394)
+  local dep_flag= dep_module=
+  if builder_is_dep_build; then
+    dep_flag=--builder-dep-parent
+    dep_module="$builder_dep_parent"
+  fi
+
+  "$script" \
     --builder-child \
     $_builder_build_deps \
+    $dep_flag "$dep_module" \
+    $action \
```

Comment:
The action has to come after the --builder-child flag so that we can ignore if we are in --builder-child mode.